beacon-chain/db/kv/attestations.go

```diff
@@ -106,81 +135,67 @@ func (k *Store) SaveAttestations(ctx context.Context, atts []*ethpb.Attestation)
 		if err != nil {
 			return err
 		}
-		key, err := generateAttestationKey(atts[i])
+		key, err := ssz.HashTreeRoot(atts[i].Data)
 		if err != nil {
 			return err
 		}
 		encodedValues[i] = enc
-		keys[i] = key
+		keys[i] = key[:]
 	}
 	return k.db.Batch(func(tx *bolt.Tx) error {
-		bucket := tx.Bucket(attestationsBucket)
+		bkt := tx.Bucket(attestationsBucket)
 		for i := 0; i < len(atts); i++ {
```

Comment:
Would it be faster to have one loop over the attestations? It seems that you loop over it twice

Reply:
The alternative is to do the hash tree roots inside of the db tx, which is an expensive operation given the tx acquires a db lock